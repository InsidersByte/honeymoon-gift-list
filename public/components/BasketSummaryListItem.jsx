/* @flow */

import React from 'react';
import { IconButton, RaisedButton } from 'material-ui';
import Add from 'material-ui/svg-icons/content/add';
import Remove from 'material-ui/svg-icons/content/remove';
import Delete from 'material-ui/svg-icons/content/clear';
import css from './BasketSummaryListItem.styl';

type PropsType = {
    item: {
        name: string,
        imageUrl: string,
        price: number,
        quantity: number,
        total: number,
        remaining: number,
    },
    addToBasket: Function,
    removeFromBasket: Function,
    deleteFromBasket: Function,
};

const styles = {
    button: {
        minWidth: 24,
        height: 24,
    },
    icon: {
        width: 18,
        height: 18,
    },
};

export default class BasketSummaryListItem extends React.Component {
    props: PropsType;

    onAdd = () => {
        this.props.addToBasket(this.props.item);
    };

    onRemove = () => {
        this.props.removeFromBasket(this.props.item);
    };

    onDelete = () => {
        this.props.deleteFromBasket(this.props.item);
    };

    render() {
        const { item: { name, imageUrl, price, quantity, remaining } } = this.props;

        console.log(this.props.item);

        return (
            <div className={css.root}>
                <img className={css.avatar} src={imageUrl} alt={name} />

                <div className={css.textContainer}>
                    <h3 className={css.name}>{name}</h3>

                    <div>
                        <RaisedButton
                            primary
                            icon={<Remove style={styles.icon} />}
                            onClick={this.onRemove}
                            disabled={quantity === 1}
                            style={styles.button}
                        />

                        <span className={css.quantity}>
                            {quantity}
                        </span>

                        <RaisedButton
                            primary
                            icon={<Add style={styles.icon} />}
                            onClick={this.onAdd}
                            disabled={quantity === remaining}
                            style={styles.button}
                        />
                    </div>

                    <h4 className={css.total}>Â£{price * quantity}</h4>
                </div>

                <IconButton touch onClick={this.onDelete}>
                    <Delete />
                </IconButton>
            </div>
        );
    }
}
