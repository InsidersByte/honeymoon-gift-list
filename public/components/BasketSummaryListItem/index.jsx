/* @flow */

import React from 'react';
import { IconButton, RaisedButton } from 'material-ui';
import Add from 'material-ui/svg-icons/content/add';
import Remove from 'material-ui/svg-icons/content/remove';
import Delete from 'material-ui/svg-icons/content/clear';
import type { ItemType } from '../../types';
import css from './index.styl';

type PropsType = {
    item: ItemType,
    addToBasket: (item: ItemType) => void,
    removeFromBasket: (item: ItemType) => void,
    deleteFromBasket: (item: ItemType) => void,
};

const styles = {
    button: {
        minWidth: 24,
        height: 24,
    },
    icon: {
        width: 18,
        height: 18,
    },
};

const BasketSummaryListItem = ({ item, addToBasket, removeFromBasket, deleteFromBasket }: PropsType) => {
    const handleAdd = () => addToBasket(item);
    const handleRemove = () => removeFromBasket(item);
    const handleDelete = () => deleteFromBasket(item);

    const { name, imageUrl, price, quantity, remaining } = item;

    return (
        <div className={css.root}>
            <img className={css.avatar} src={imageUrl} alt={name} />

            <div className={css.textContainer}>
                <h3 className={css.name}>{name}</h3>

                <div>
                    <RaisedButton
                        primary
                        icon={<Remove style={styles.icon} />}
                        onClick={handleRemove}
                        disabled={quantity === 1}
                        disableFocusRipple
                        disableKeyboardFocus
                        disableTouchRipple
                        style={styles.button}
                    />

                    <span className={css.quantity}>
                        {quantity}
                    </span>

                    <RaisedButton
                        primary
                        icon={<Add style={styles.icon} />}
                        onClick={handleAdd}
                        disabled={quantity === remaining}
                        disableFocusRipple
                        disableKeyboardFocus
                        disableTouchRipple
                        style={styles.button}
                    />
                </div>

                <h4 className={css.total}>Â£{price * quantity}</h4>
            </div>

            <IconButton touch onClick={handleDelete}>
                <Delete />
            </IconButton>
        </div>
    );
};

export default BasketSummaryListItem;
