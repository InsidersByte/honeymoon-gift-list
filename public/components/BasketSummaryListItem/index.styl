$avatar-width = 60px
$top-bottom-margin = 12px

.root
    align-items center
    display flex
    padding 0 16px

.avatar
    border-radius 50%
    height $avatar-width
    margin $top-bottom-margin 16px $top-bottom-margin 0
    width $avatar-width

.textContainer
    flex 1
    margin-right 16px

.name
    font-size 1.2em
    margin $top-bottom-margin 0

.total
    font-size 1em

.quantity
    margin 0 10px
