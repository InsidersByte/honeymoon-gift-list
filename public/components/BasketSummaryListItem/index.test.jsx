import React from 'react';
import { shallow } from 'enzyme';
import BasketSummaryListItem from './';

const item = { id: 2, price: 2, quantity: 2, remaining: 20 };

const addToBasket = jest.fn();
const removeFromBasket = jest.fn();
const deleteFromBasket = jest.fn();

const props = {
    item,
    addToBasket,
    removeFromBasket,
    deleteFromBasket,
};

describe('BasketSummaryListItem', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <BasketSummaryListItem {...props} />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should handle deleteFromBasket correctly', () => {
        const wrapper = shallow(
            <BasketSummaryListItem {...props} />,
        );

        const deleteButton = wrapper.find('IconButton');

        expect(deleteButton.length).toBe(1);
        expect(deleteFromBasket).not.toHaveBeenCalled();

        deleteButton.simulate('click');

        expect(deleteFromBasket).toHaveBeenCalledTimes(1);
        expect(deleteFromBasket).toHaveBeenCalledWith(item);
    });
});
