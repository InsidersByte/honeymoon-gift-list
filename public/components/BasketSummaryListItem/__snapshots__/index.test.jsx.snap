// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketSummaryListItem should render correctly 1`] = `
<div
  className="root"
>
  <img
    className="avatar"
  />
  <div
    className="textContainer"
  >
    <h3
      className="name"
    />
    <div>
      <RaisedButton
        disableFocusRipple={true}
        disableKeyboardFocus={true}
        disableTouchRipple={true}
        disabled={false}
        fullWidth={false}
        icon={
          <ContentRemove
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
        }
        labelPosition="after"
        onClick={[Function]}
        primary={true}
        secondary={false}
        style={
          Object {
            "height": 24,
            "minWidth": 24,
          }
        }
      />
      <span
        className="quantity"
      >
        2
      </span>
      <RaisedButton
        disableFocusRipple={true}
        disableKeyboardFocus={true}
        disableTouchRipple={true}
        disabled={false}
        fullWidth={false}
        icon={
          <ContentAdd
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
        }
        labelPosition="after"
        onClick={[Function]}
        primary={true}
        secondary={false}
        style={
          Object {
            "height": 24,
            "minWidth": 24,
          }
        }
      />
    </div>
    <h4
      className="total"
    >
      Â£
      4
    </h4>
  </div>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onClick={[Function]}
    tooltipPosition="bottom-center"
    touch={true}
  >
    <ContentClear />
  </IconButton>
</div>
`;
